crates/turborepo-lib/src/opts.rs

```diff
@@ -362,16 +358,37 @@ impl<'a> TryFrom<OptsInputs<'a>> for ScopeOpts {
 impl<'a> From<OptsInputs<'a>> for CacheOpts {
     fn from(inputs: OptsInputs<'a>) -> Self {
         let is_linked = turborepo_api_client::is_linked(inputs.api_auth);
-        let skip_remote = if !is_linked {
-            true
+        let mut cache = inputs.config.cache();
```

Comment:
I think this layering logic is confusing since env vars or config settings can override CLI level cache settings:
e.g. `TURBO_FORCE=1 turbo run build --force=false` would result in `remote:rw,local:rw` cache behavior, but `TURBO_REMOTE_ONLY=1 turbo run build --cache='local:rw` would result in no caches are enabled.

I think it would make more sense to fold the existing cache flags into the new cache struct at each level in the config, but I'm not convinced that wouldn't be confusing. Could we get away with having the new cache struct conflict with existing cache flags at each level e.g `turbo run build --remote-only --cache='...'` would throw but `TURBO_REMOTE_ONLY=1 turbo run build --cache='...'` be fine where `--cache` wins out completely?

Reply:
Is there a way to have a group of flags conflict with a single flag but not with each other? I wasn't sure if that's possible in clap. Ofc I can manually add that as a restriction

Reply:
The layering issue is a tricky one. If the config specifies `remote:rw` in a file, but then we add on `local:rw`, do we want to do an `OR` or an `AND` for those? I could see either one working